Wikipedia/Code/WMFAppViewController.m

```diff
@@ -994,6 +994,18 @@ - (void)pauseApp {
         [self.dataStore removeArticlesWithURLsFromCache:articleURLsToRemoveFromDisk];
     }
 
+    // TODO: instead of checking `savedPageList` just query the data store for articles with the isConversionFromMobileviewNeeded flag true?
```

Comment:
Yes, and it should operate in batches, most likely only one article at a time. With this setup, it looks like the `WKWebView` in `MobileviewToMobileHTMLConverter` often doesn't load the page when the app is backgrounded and waits til the next foregrounding. This delay causes the migrations in the buffer to remain incomplete until the next launch. Given this, it's likely this queue would be better off in `SavedArticlesFetcher`(which is really a `SavedArticlesController` in our current naming convention) and operate in a similar manner as the download queue - fetch and convert one at a time while the app is foregrounded.